devices/Ds18b20/Ds18b20.cs

```diff
@@ -0,0 +1,540 @@
+        /// If ManyDevicesConnected is set it will keep searching until find last device, saving all in AddressNet array.
+        /// </summary>
+        /// <returns><b>true</b> on success, else <b>false</b></returns>
+        public bool Initialize()
+        {
+            int foundDevices = 0;
+            ArrayList allDevices = new ArrayList();
+
+            _oneWireHost.TouchReset();
+
+            if (Address == null)
+            {
+                if (_manyDevicesConnected)
+                {
+                    if (_oneWireHost.FindFirstDevice(false, IsAlarmSearchCommandEnabled))
+                    {
+                        do
+                        {
+                            if (_oneWireHost.SerialNumber[0] == (byte)RomCommands.FamilyCode)
+                            {
+                                _oneWireHost.TouchReset();
+                                Address = new byte[_oneWireHost.SerialNumber.Length];
+                                Array.Copy(_oneWireHost.SerialNumber, Address, _oneWireHost.SerialNumber.Length);
+                                foundDevices++;
+                                allDevices.Add(Address);
+                            }
+                        }
+                        while (_oneWireHost.FindNextDevice(false, IsAlarmSearchCommandEnabled));
+                    }
+
+                    if (foundDevices > 0)
+                    {
+                        AddressNet = new byte[foundDevices][];
+                        int i = 0;
+                        foreach (byte[] device in allDevices)
+                        {
+                            AddressNet[i] = new byte[device.Length];
+                            Array.Copy(device, AddressNet[i], device.Length);
+                            i++;
+                        }
+                    }
+                }
+                else
+                {
+                    if (_oneWireHost.FindFirstDevice(true, IsAlarmSearchCommandEnabled))
+                    {
+                        do
+                        {
+                            if (_oneWireHost.SerialNumber[0] == (byte)RomCommands.FamilyCode)
+                            {
+                                Address = new byte[_oneWireHost.SerialNumber.Length];
+                                Array.Copy(_oneWireHost.SerialNumber, Address, _oneWireHost.SerialNumber.Length);
+                                foundDevices = 1;
+                                break;
+                            }
+                        }
+                        while (_oneWireHost.FindNextDevice(true, IsAlarmSearchCommandEnabled));
+                    }
+                }
+            }
+
+            return foundDevices > 0;
+        }
+
+        /// <summary>
+        /// Read sensor data.
+        /// </summary>
+        /// <returns><b>true</b> on success, else <b>false</b></returns>
+        public bool Read()
```

Comment:
So as there is a function like this to read the sensor, you can as well one with the `bool TryRead(out Temperature temp)` which is agood pattern to add for those kind of sensors (there are example like the DHT, the BMP and few others already). Just add this and should be good. So the pattern of trying to read can be used.